src/components/FamilyManifestModal/FamilyDataTypesStatsQuery.js

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+
+import { get, difference } from 'lodash';
+import Query from '@arranger/components/dist/Query';
+import { withApi } from 'services/api';
+import { compose } from 'recompose';
+
+export default compose(withApi)(
+  ({ api, dataTypes = [], participantIds, projectId, isDisabled, render = () => {} } = {}) => (
+    <Query
+      renderError
+      api={api}
+      projectId={projectId}
+      name={`dataTypeQuery`}
+      query={`
+        query dataTypes(${dataTypes.map((dataType, i) => `$sqon${i}: JSON`).join(', ')}) {
+          file {
+            ${dataTypes
+              .map(
+                (dataType, i) => `
+                ${dataType.key.replace(/[^\da-z]/gi, '')}: aggregations(filters: $sqon${i}) {
+                  size {
+                    stats {
+                      sum
+                    }
+                  }
+                }
+                ${dataType.key.replace(/[^\da-z]/gi, '')}family: aggregations(filters: $sqon${i}) {
+                  participants__family__family_compositions__family_members__kf_id {
+                    buckets {
+                      key
+                    }
+                  }
+                }
+              `,
+              )
+              .join('\n')}
+          }
+        }
+      `}
+      variables={dataTypes.reduce((acc, dataType, i) => {
+        const dataTypeFilters = ids => [
+          {
+            op: 'in',
+            content: { field: 'data_type', value: [dataType.key] },
+          },
+          {
+            op: 'in',
+            content: { field: 'participants.kf_id', value: ids },
+          },
+        ];
+
+        return {
+          ...acc,
+          [`sqon${i}`]: {
+            op: 'and',
+            content: dataTypeFilters(participantIds),
+          },
+        };
+      }, {})}
+      render={({ data, loading }) => {
+        return render(
+          loading
+            ? { loading }
```

Comment:
might be helpful to return a default data empty array here to guard against null exceptions for consumers?
```
return render({ loading, data: loading ? [] : dataTypes.map(...) })
```
?